File: Task 1/C++/Components/Product/Decor.cpp
#include "Decor.h"

Decor::Decor() : Furniture(){
    style = "";
}

Decor::Decor(std::string name, float cost, std::string supplier, float discount_amount, int packing_units, float cost_per_unit, std::string material, std::string style)
        : Furniture(name, cost, supplier, discount_amount, packing_units, cost_per_unit, material){
    this->style = style;
}

Decor::Decor(const Decor& other) : Furniture(other) {
    this->style = other.style;
}


std::string Decor::display_product_info(){
    return Furniture::display_product_info() +
           "\nStyle: " + style;
}

File: Task 1/Java/logistics/src/main/java/com/logistics/Components/Product/Decor.java
package com.logistics.Components.Product;

public class Decor extends Furniture {

    private String style;

    public Decor() {
        super();
        style = "";
    }

    public Decor(String name, float cost, String supplier, float discountAmount, int packingUnits, float costPerUnit, String material, String style) {
        super(name, cost, supplier, discountAmount, packingUnits, costPerUnit, material);
        this.style = style;
    }

    // Copy Constructor
    public Decor(Decor other) {
        super(other);
        this.style = other.style;
    }

    // Clone method
    @Override
    public Decor clone() {
        return new Decor(this);
    }

    // Getters
    public String getStyle() { return style; }

    // Setters
    public void setStyle(String style) { this.style = style; }

    @Override
    public String display_product_info() {
        return super.display_product_info() + 
                "\nStyle: " + style;
    }
}
